Security/Firewall/AbstractOAuthListener.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the FOSOAuthServerBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\OAuthServerBundle\Security\Firewall;
+
+use FOS\OAuthServerBundle\Security\Authentication\Token\OAuthToken;
+use OAuth2\OAuth2;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Security\Core\Authentication\AuthenticationManagerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+use Symfony\Component\Security\Http\Firewall\ListenerInterface;
+
+/**
+ * AbstractOAuthListener class.
+ *
+ * @author Arnaud Le Blanc <arnaud.lb@gmail.com>
+ */
+abstract class AbstractOAuthListener implements ListenerInterface
```

Comment:
please don't use 2 classes. Instead, make the OAuthListener class accept both SecurityContextInterface and TokenStorageInterface as first argument (it is easy given that the way to set the token is the same in both cases anyway)